tar/bsdtar.c

```diff
@@ -1687,16 +1702,33 @@ dooption(struct bsdtar *bsdtar, const char * conf_opt,
 	    "Unrecognized configuration file option: \"%s\"", conf_opt);
 }
 
-/* Load keys from the specified file. */
-static void
-load_keys(struct bsdtar *bsdtar, const char *path)
+/*
+ * Load keys from the specified file.  Return success or file-not-found; exits
+ * directly if a non-recoverable failure is encountered.
+ */
+static int
+load_keys(struct bsdtar *bsdtar, const char *path, int fromconffile)
 {
 	uint64_t machinenum;
+	int rc = 1;
 
 	/* Load the key file. */
-	if (keyfile_read(path, &machinenum, ~0))
+	switch (keyfile_read(path, &machinenum, ~0)) {
+	case 0:
+		break;
+	case 1: /* file not found */
+		if (fromconffile) {
+			bsdtar_warnc(bsdtar, errno,
```

Comment:
Won't this result in "blahblahblah continuing anyway" immediately followed by a complaint about missing keys if we try to perform a non-dryrun when our config file points at a nonexistent key file?